web/src/beta/components/fields/URLField/index.tsx

```diff
@@ -0,0 +1,175 @@
+import React, { useCallback, useState } from "react";
+
+import Button from "@reearth/beta/components/Button";
+import Property from "@reearth/beta/components/fields";
+import TextInput from "@reearth/beta/components/fields/common/TextInput";
+import useHooks from "@reearth/beta/features/Assets/AssetsQueriesHook/hooks";
+import {
+  FILE_FORMATS,
+  IMAGE_FORMATS,
+  VIDEO_FORMATS,
+} from "@reearth/beta/features/Assets/constants";
+import { Asset } from "@reearth/beta/features/Assets/types";
+import { useManageAssets } from "@reearth/beta/features/Assets/useManageAssets/hooks";
+import ChooseAssetModal from "@reearth/beta/features/Modals/ChooseAssetModal";
+import { checkIfFileType } from "@reearth/beta/utils/util";
+import { useT } from "@reearth/services/i18n";
+import { useNotification, useWorkspace } from "@reearth/services/state";
+import { styled } from "@reearth/services/theme";
+
+export type Props = {
+  value?: string;
+  onChange?: (value: string | undefined) => void;
+  name?: string;
+  description?: string;
+  fileType?: "asset" | "URL";
+  assetType?: "image" | "file";
+};
+
+const URLField: React.FC<Props> = ({ name, description, value, fileType, assetType, onChange }) => {
+  const t = useT();
+  const [open, setOpen] = useState(false);
+  const [currentWorkspace] = useWorkspace();
+  const [, setNotification] = useNotification();
+  const [CurrentValue, setCurrentValue] = useState(value);
+  const {
+    assets,
+    isLoading,
+    hasMoreAssets,
+    searchTerm,
+    selectedAssets,
+    selectAsset,
+    handleGetMoreAssets,
+    createAssets,
+    handleSortChange,
+    handleSearchTerm,
+  } = useHooks({ workspaceId: currentWorkspace?.id });
+
+  const {
+    localSearchTerm,
+    wrapperRef,
+    onScrollToBottom,
+    sortOptions,
+    handleSearchInputChange,
+    handleUploadToAsset,
+    handleSearch,
+  } = useManageAssets({
+    selectedAssets,
+    searchTerm,
+    isLoading,
+    hasMoreAssets,
+    onGetMore: handleGetMoreAssets,
+    onSortChange: handleSortChange,
+    onCreateAssets: createAssets,
+    onSearch: handleSearchTerm,
+  });
+
+  const handleClose = useCallback(() => {
+    setOpen(false);
+  }, []);
+
+  const handleClick = useCallback(() => setOpen(!open), [open]);
+
+  const handleSelect = useCallback(
+    (asset?: Asset) => {
+      if (!asset) return;
+      if (!selectedAssets.includes(asset)) selectAsset([asset]);
+      else selectAsset([]);
+    },
+    [selectedAssets, selectAsset],
+  );
+
+  const handleUpload = useCallback(() => {
+    handleUploadToAsset();
+  }, [handleUploadToAsset]);
```

Comment:
Since nothing really is going on here just use handleUploadToAsset directly

Reply:
still working on this